src/services/ElasticsearchService.js

```diff
@@ -5,15 +5,15 @@ import {
 import AwsConnection from './aws/AwsElasticsearchConnection';
 
 class ElasticsearchService {
-  constructor({ index, type, connection, options }) {
```

Comment:
May I ask to expound more on what you mean by new Service for Search specific? Also, I think passing of a custom client is consistent with Dependency Injection (an oop concept). This allows the changes on this public facing code to be as minimal as possible. A helper factory function is also provided in the plugin which helps create the client object easier.

Reply:
Yep this is technically correct, but by concept yes you are doing a helper rather than a service. Services usually don't need to have the dependency injection, though yes it's easier and more minimal change, but this will make it now as a service that would have very minimal flexibiltiy.

The reason for this is say in Laravel, they chose to instead e.g. have you pass a client. They make you instead declare e.g. MysqlConnector or DynamodbConnector or Mongodbconnector all as separate services even facades (or utils in Lesgo). In fact, they can just pass in client like you did but they didn't it's for the same reason